crates/circuit/src/dag_circuit.rs

```diff
@@ -2414,78 +2415,72 @@ def _format(operand):
-        //         new_wires.update(node_resources(op.target).clbits)
-        //
-        // if propagate_condition and not (
-        //     isinstance(node.op, ControlFlowOp) or isinstance(op, ControlFlowOp)
-        // ):
-        //     if new_condition is not None:
-        //         raise DAGCircuitError(
-        //             "Cannot propagate a condition to an operation that already has one."
-        //         )
-        //     if (old_condition := getattr(node.op, "condition", None)) is not None:
-        //         if not isinstance(op, Instruction):
-        //             raise DAGCircuitError("Cannot add a condition on a generic Operation.")
-        //         if not isinstance(node.op, ControlFlowOp):
-        //             op = op.c_if(*old_condition)
-        //         else:
-        //             op.condition = old_condition
-        //         new_wires.update(condition_resources(old_condition).clbits)
-        //
-        // if new_wires != current_wires:
-        //     # The new wires must be a non-strict subset of the current wires; if they add new wires,
-        //     # we'd not know where to cut the existing wire to insert the new dependency.
-        //     raise DAGCircuitError(
-        //         f"New operation '{op}' does not span the same wires as the old node '{node}'."
-        //         f" New wires: {new_wires}, old wires: {current_wires}."
-        //     )
-        //
-        // if inplace:
-        //     if op.name != node.op.name:
-        //         self._increment_op(op)
-        //         self._decrement_op(node.op)
-        //     node.op = op
-        //     return node
-        //
-        // new_node = copy.copy(node)
-        // new_node.op = op
-        // self._multi_graph[node._node_id] = new_node
-        // if op.name != node.op.name:
-        //     self._increment_op(op)
-        //     self._decrement_op(node.op)
-        // return new_node
-        todo!()
+    ) -> PyResult<Py<PyAny>> {
+        // 1. look up dag weight associated to node -> ref
+        let node_id = node.as_ref().node.unwrap();
+        let node_weight = self.dag.node_weight(node_id);
+        // 2. get information from new op
+        let op_construct = convert_py_to_operation_type(op.py(), op.clone().unbind())?;
+        // 3. get value of "condition"
+        let condition = match propagate_condition
+            && !op_construct.operation.control_flow()
+            && !node.instruction.operation.control_flow()
+        {
+            true => match &node.instruction.extra_attrs {
+                Some(extra_attrs) => &extra_attrs.condition,
+                None => &None,
+            },
+            _ => &op_construct.condition,
+        };
+        // 4. update node_weight with op info
+        match node_weight {
+            Some(NodeType::Operation(packed)) => {
+                // how do we get a mutable reference to packed?
+                packed.op = op_construct.operation.into();
```

Comment:
I am not sure how to get a mutable reference to `packed` so that we can replace the node info, as it comes from `node_weight` which is returned immutably from `self.dag.node_weight(node_id)`